test/distributed/test_c10d.py

```diff
@@ -355,54 +353,42 @@ def test_compare_set(self):
         self.assertEqual(b"new_value0", new_value_result)
         self.assertEqual(b"new_value0", store.get("key0"))
 
-    def _create_client(self, addr, port, world_size):
+    def _create_client(self, addr, port, world_size, messages):
         try:
             client_store = dist.TCPStore(addr, port, world_size, timeout=timedelta(seconds=10))
             self.assertEqual(b"value", client_store.get("key"))
             client_store.set("new_key", "new_value0")
             client_store.compare_set("new_key", "new_value1", "new_value0")
         except Exception:
-            logging.error('Caught exception: \n{}exiting process with exit code: {}'
-                          .format(traceback.format_exc(), MultiProcessTestCase.TEST_ERROR_EXIT_CODE))
+            messages.put('Caught exception: \n{}exiting process with exit code: {}'
+                         .format(traceback.format_exc(), MultiProcessTestCase.TEST_ERROR_EXIT_CODE))
             sys.exit(MultiProcessTestCase.TEST_ERROR_EXIT_CODE)
 
-    @unittest.skipIf(IS_WINDOWS, "Skip test for windows because it cannot serialize logger instance")
-    def test_multi_worker_with_fixed_world_size(self):
+    def _multi_worker_helper(self, world_size):
         addr = DEFAULT_HOSTNAME
-        port = common.find_free_port()
-        world_size = 5
+        server_store = create_tcp_store(addr, world_size, wait_for_workers=False)
+        server_store.set("key", "value")
+        port = server_store.port
+        messages = mp.Queue()
         processes = []
-        for _ in range(world_size):
-            p = mp.Process(target=self._create_client, args=(addr, port, world_size))
+        num_proccesses = random.randint(3, 5) if world_size == -1 else world_size
+        for _ in range(num_proccesses):
+            p = mp.Process(target=self._create_client, args=(addr, port, world_size, messages))
             processes.append(p)
             p.start()
-        server_store = dist.TCPStore(addr, port, world_size, True, timedelta(seconds=30))
-        server_store.set("key", "value")
-        for (i, p) in enumerate(processes):
-            # This is the exit code processes exit with if they encountered an exception.
-            self.assertNotEqual(p.exitcode, MultiProcessTestCase.TEST_ERROR_EXIT_CODE, 
-                                "Process {} terminated with exit code {}. Check logs for exception stacktrace."
-                                .format(i, p.exitcode))
+        for p in processes:
             p.join()
+        error_message = ""
+        while not messages.empty():
+            error_message += messages.get() + "\n"
+        if error_message:
```

Comment:
A process might also fail abruptly without a clean path (e.g. OOM, SEGV). I suggest checking the exit codes instead of the length of the `error_message` here.

Reply:
Thanks! I updated the check